src/coreclr/jit/emitarm64.cpp

```diff
@@ -11047,63 +11104,24 @@ void emitter::emitIns_R_R_R_I(instruction ins,
             break;
 
         case INS_sve_cdot:
-            assert(insOptsScalableStandard(opt));
-            assert(isVectorRegister(reg1)); // ddddd
-            assert(isVectorRegister(reg2)); // nnnnn
-            assert(isVectorRegister(reg3)); // mmm
-            if (opt == INS_OPTS_SCALABLE_B)
-            {
-                assert(isValidUimm4(imm)); // ii rr
-                assert((REG_V0 <= reg3) && (reg3 <= REG_V7));
-                fmt = IF_SVE_FA_3A;
```

Comment:
what are these changes? I see that we moved it under `emitIns_R_R_R_I_I()` and wondering why this was not done when we implemented `SVE_FA_3A`, `SVE_FA_3B`, etc. ?

Reply:
When I first added `emitIns_R_R_R_I_I`, I was trying to minimize code duplication, so I just made it a wrapper for `emitIns_R_R_R_I` by bitwise OR-ing `imm1` and `imm2` into one `imm`, and then passing this along to `emitIns_R_R_R_I` to do the rest. Now I'm running into instructions that have encodings that take one immediate, and encodings that take two immediates, so it's easier to separate these two `emitIns` methods out.